### PR TITLE
Feature/cloud 1751 pin version numbers

### PR DESCRIPTION
**Pull request by naveen-variant**
# Description

Pin version numbers to terraform modules

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change
(fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

```yaml
Pre reqs:
  - Create a branch in demo flask api and add postgress database https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml#L10
  - Create a commit and deploy (sample: https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/demo-test-postgress/deployments/releases/0.1.0-test-master-post0001.89/deployments/Deployments-38114?activeTab=taskLog)

Test Case 1 [Verify app is deployed and nothing is breaking]:
- Result
 - App should be deployed in cluster

Test Case 2 [See database is created and run read queries with the role]:
- Steps
  - run /database/read
- Results
  - Should return true as response
  - Check pod logs for results from select comand

Test Case 2 [See database is created and run write queries with the role]:
- Steps
  - run /database/write
- Results
  - Should return true as response
  - Check pod logs for results from table creation and insert results
  
Test Case 3 [Delete the records insert]:
- Steps
  - run /database/delete
- Results
  - Should return true as response
  - Check pod logs for results from table and records deletion

Test Case 4 [To see tags version pinning is working]:
- Steps
  - Make sure app is deployed
- Results
  - check in app role tags to see tags are added as given in api.yaml

Test Case 5 [To create s3 bucket]:
- Steps
  - Make sure app is deployed
- Results
  - check bucket is created

Test Case 6 [To create SNS Topic]:
- Steps
  - Make sure app is deployed
  - With the given sns topic spec https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml
- Results
  - Should see sns got created.

Test Case 7 [To create SNS/SQS subscription]:
- Steps
  - Make sure app is deployed
  - With the given sns topic spec https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml
- Results
  - Should see sqs got created.
  - Should see sns/sqs subscription got created.
 
Test Case 7 [Run SNS/SQS endpoints]:
- Steps
  - Make sure app is deployed
  - With the given sns topic spec https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml
  - run /sns endpoint
  - run /sqs endpoint(Should add some message to external topic so that sqs can pull that message after subscription)

- Results
  - Should see message got posted in the pod logs.
  - Expect some some delay to receive response after post message to externat topic , when /sqs endpoint has ran

 
Test Case 8 [Check Dlq created]:
- Steps
  - Make sure app is deployed
  - With the given sns topic spec https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml
- Results
  - Check dlq is created

Test Case 9 [Deploy Dynamo db/ run endpoint]:
- Steps
  - Make sure app is deployed
  - With the given dynamo db spec https://github.com/variant-inc/demo-python-flask-variant-api/tree/test/dynamodb
  - octopus link https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/demo-test-dynamodb/deployments
  - run /dynamo-insert endpoint
- Results
  - Should see table got created and record got inserted.
```

- [ ] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by vibindaniel — DISMISSED** [review]

File: modules/common/postgres/main.tf [hunk 1]
```diff
@@ -12,7 +12,7 @@ locals {
 
 module "database" {
   for_each        = local.database_map
-  source          = "github.com/variant-inc/terraform-postgres-database?ref=v1"
+  source          = "github.com/variant-inc/terraform-postgres-database?ref=v0.1.0"
```
[the remark this answers is not in the document]
Comment (naveen-variant) [reply modules/common/postgres/main.tf]: That was from lazy terraform. Got fixed after pointing to right version

File: live/common/tags/terragrunt.hcl [hunk 2]
```diff
@@ -7,7 +7,7 @@ locals {
 }
 
 terraform {
-  source = "github.com/variant-inc/lazy-terraform//submodules/tags?ref=v1"
+  source = "github.com/variant-inc/lazy-terraform//submodules/tags?ref=v1.9.0"
```
[the remark this answers is not in the document]
Comment (naveen-variant) [reply live/common/tags/terragrunt.hcl]: Done

**Review by LeeHughsVariant — APPROVED** [review]

File: modules/common/buckets/managed.tf [hunk 3]
```diff
@@ -6,7 +6,7 @@ locals {
 # Create the buckets that are managed and owned by this app
 module "buckets" {
   for_each                   = local.managed_map
-  source                     = "github.com/variant-inc/terraform-aws-s3.git?ref=v1.2.0"
+  source                     = "github.com/variant-inc/terraform-aws-s3.git?ref=v1.2"
```
Comment (rchengvariant) [inline modules/common/buckets/managed.tf:9]: This should be `1.3`.

Comment (naveen-variant) [reply modules/common/buckets/managed.tf:9]: We are not pointing to latest version yet

**Pull request merged**